tutorials/persistance-comparison-series/BinaryWriterReader/binary_reader_writer.md

```diff
@@ -0,0 +1,426 @@
+    // They need to be disposed at the end, so `using` is good practice
+    // because it does this automatically.
+    using FileStream fileStream = File.Open(HitCountFile, FileMode.Create); // 11
+    using BinaryWriter binaryWriter = new(fileStream, Encoding.UTF8); // 12
+    binaryWriter.Write(hitCountUnmodified); // 13
+    binaryWriter.Write(hitCountShift); // 13
+    binaryWriter.Write(hitCountControl); // 13
+}
+```
+
+Whenever a mouse click is detected on the capsule (6), we can then perform a similar check to what happened in `Update()`, only we use `modifier` instead of `Input.GetKey()` here.
+
+Check if `modifier` was set to `KeyCode.LeftShift` or `KeyCode.LeftControl` (7) and if so, increment the corresponding hit count (8). If no modifier was used (9), increment the `hitCountUnmodified` (10).
+
+Similar to the simple version, we create a `FileStream` (11) and with it the `BinaryWriter` (12). Writing multiple variables into the file can simply be achieved by calling `Write()` multiple times (13), once for each hit count that we want to save.
+
+Start the game, and click the capsule using Shift and Control. You should see the three counters in the Inspector.
+
+<img src="images/07_hit_count_extended_editor.jpg" alt="File BinaryReaderWriterExampleExtended" width=""/>
+
+After stopping the game and therefore saving the data, a new file `BinaryReaderWriterExampleExtended` should exist in your project folder. Have a look at it. It should look something like this:
+
+<img src="images/08_hit_count_extended_file.png" alt="File BinaryReaderWriterExampleExtended" width=""/>
+
+The three hit counters can be seen in there and correspond to the values in the inspector:
+
+- `0f` == 15
+- `0c` == 12
+- `05` == 5
+
+Last but not least, let's look at how to load the file again when starting the game (14):
+
+```cs
+private void Start() // 14
+{
+    // Check if the file exists to avoid errors when opening a non-existing file.
+    if (File.Exists(HitCountFile)) // 15
+    {
+        // Open a stream to the file that the `BinaryReader` can use to read data.
+        // They need to be disposed at the end, so `using` is good practice
+        // because it does this automatically.
+        using FileStream fileStream = File.Open(HitCountFile, FileMode.Open); // 16
+        using BinaryReader binaryReader = new(fileStream); // 17
+        hitCountUnmodified = binaryReader.ReadInt32(); // 18
+        hitCountShift = binaryReader.ReadInt32(); // 18
+        hitCountControl = binaryReader.ReadInt32(); // 18
+    }
+}
+```
+
+First, we check if the file even exists (15). If we ever saved data before, this should be the case. If it exists, we read the databy creating a `FileStream` again (16) and opening a `BinaryReader` with it (17). Similar to writing with `Write()` (on the `BinaryWriter`), we use `ReadInt32()` (18) to read an `integer`. We do this three times since we saved them all individually.
```

Comment:
```suggestion
First, we check if the file even exists (15). If we ever saved data before, this should be the case. If it exists, we read the data by creating a `FileStream` again (16) and opening a `BinaryReader` with it (17). Similar to writing with `Write()` (on the `BinaryWriter`), we use `ReadInt32()` (18) to read an `integer`. We do this three times since we saved them all individually.
```

Reply:
ac15b8e9decad9f95666d5e37b811f651dc4a7dc